### PR TITLE
enhance/search

### PR DESCRIPTION
**Pull request by DmitriVanGuard**
### Summary
Implementing new search architecture.
- Displaying recently searched assets on empty input focus;
- Filtered assets are sorted by the `rank` field;
- Displaying `Trending words` category that consists of the **top 10** words for a past 8 hours sorted by the `score`;
- After selecting item from:
  - `Assets` category, it will be added to the `Recently searched` category and user will be redirected to the `/projects/:suggestion` page
  - `Trending words` category, user will be redirected to the `/labs/trends/explore/:suggestion` page.

### Screenshots
gif

### CODE REVIEW

File: src/components/Search/SearchContainer.js [hunk 1]
```diff
@@ -1,25 +1,84 @@
...
       className={cx(styles.wrapper, className)}
       iconPosition='left'
-      onSuggestionSelect={({ slug }) => history.push(`/projects/${slug}`)}
+      onSuggestionSelect={({ category, item }) => {
+        if (category === ASSETS || category === RECENT_ASSETS) {
+          const { slug = item } = item
+          addRecentAssetSuggestions(slug)
+          history.push(`/projects/${slug}`)
+        } else if (category === TRENDING_WORDS) {
+          history.push(`/labs/trends/explore/${item}`)
```
Comment (GarageInc) [inline src/components/Search/SearchContainer.js]: What will be a right way: dispatch(push(path)) or history.push(path)?

File: src/components/Search/SearchProjects.js [hunk 2]
```diff
@@ -8,65 +10,131 @@ import ProjectIcon from './../ProjectIcon'
...
+        {
+          predicate: assetsPredicate,
+          title: 'Assets',
+          items: projects,
+          suggestionContent: ({ name, ticker, id }) => {
+            const isAssetInList = isEditingWatchlist
+              ? hasAssetById({ listItems: watchlistItems, id })
+              : false
+            return (
+              <div className={styles.projectWrapper}>
```
Comment (GarageInc) [inline src/components/Search/SearchProjects.js]: Maybe new component for suggestionContent?

File: src/components/Search/SearchProjects.js [hunk 3]
```diff
@@ -8,65 +10,131 @@ import ProjectIcon from './../ProjectIcon'
...
-  options: () => ({
-    context: { isRetriable: true },
-    variables: { minVolume: 0 }
+const enhance = compose(
+  graphql(TRENDING_WORDS_QUERY, {
+    options: () => {
+      const from = new Date()
+      const to = new Date()
+      to.setHours(to.getHours(), 0, 0, 0)
+      from.setHours(from.getHours() - 8, 0, 0, 0)
```
Comment (Partysun) [inline src/components/Search/SearchProjects.js]: Why do we need to set up 8 hours? Not 1? @DmitriVanGuard

Comment (DmitriVanGuard) [reply src/components/Search/SearchProjects.js]: Similar approach that we have used for the `Last trends`. But sure, we can make them "more" up-to-date in a search box. I will change it to the `1 hour`

File: src/components/Search/SearchProjects.js (hunk 2 above)
Comment (DmitriVanGuard) [reply src/components/Search/SearchProjects.js]: Indeed, better to move into component

**Pull request merged**